integration-tests/src/test/java/io/druid/tests/indexer/ITKafkaIndexingServiceTest.java

```diff
@@ -184,16 +184,23 @@ public void testKafka()
         throw Throwables.propagate(ioe);
       }
 
-      try {
-        Thread.sleep(DELAY_BETWEEN_EVENTS_SECS * 1000);
-      }
-      catch (InterruptedException ex) { /* nothing */ }
       dtLast = dt;
       dt = new DateTime(zone);
     }
 
     producer.close();
 
+    long queryStartTime = System.currentTimeMillis() + WAIT_TIME_MILIIS;
+    while(queryStartTime > System.currentTimeMillis()) {
+      try {
+        Thread.sleep(5000);
+      }
+      catch (InterruptedException e) {
+        e.printStackTrace();
```

Comment:
if the thread is interrupted i guess there is a reason for that thus you need to propagate and set the interrupt flag thought.